lib/tessel/provision.js

```diff
@@ -170,6 +180,27 @@ actions.checkAuthFileExists = function(tessel, authFile) {
   });
 };
 
+actions.setDefaultKey = function(opts) {
+
+  if (opts.key && typeof opts.key === 'string') {
+    debug('Using --key ' + opts.key + ' after checking existens');
+    filepath = opts.key;
+  } else if (opts.key && typeof opts.key !== 'string') {
+    logs.err('No valid path for private key');
+    process.exit();
+  }
+  try {
+    if (fs.statSync(filepath).isFile() &&
+      fs.statSync(filepath + '.pub').isFile()) {
+      return;
```

Comment:
I don't follow your logic. It looks like the first branch of the if statement sets the key if one was provided in the options. If a key wasn't provided, it checks to make sure the current key exists? It seems like you shouldn't be able to call this function without a new key.